File: ZetCodeQt5Exmaple/MenusAndToolbars/toolbar.cpp
#include "toolbar.h"
#include <QToolBar>
#include <QIcon>
#include <QAction>
#include <QPixmap>

ToolBar::ToolBar(QWidget *parent)
    : QMainWindow(parent)
{
    QPixmap newpix(":/img/new.png");
    QPixmap openpix(":/img/open.png");
    QPixmap quitpix(":/img/quit.png");

    QToolBar *toolbar = addToolBar("main toolbar");  // create toolbar
    toolbar->addAction(QIcon(newpix), "new File");
    toolbar->addAction(QIcon(openpix), "Open File");
    toolbar->addSeparator();
    QAction *quit = toolbar->addAction(QIcon(quitpix),
                                       "Quit Application");

    connect(quit, &QAction::triggered, qApp, &QApplication::quit);
}

File: ZetCodeQt5Exmaple(Python)/menuAndToolbars/Toolbar.py
#!/usr/bin/python3
#-*- coding: utf-8 -*-

"""
This program creates a toolbar

author: xxxx
"""


import sys
from PyQt5.QtWidgets import QMainWindow, QMenu, QApplication, QAction, qApp
from PyQt5.QtGui import QIcon


class Example(QMainWindow):
    def __init__(self):
        super().__init__()
        self.initUI()

    def initUI(self):
        exitAct = QAction(QIcon('exit.png'), '退出', self)
        exitAct.setShortcut('Ctrl+Q')
        exitAct.triggered.connect(qApp.quit)

        self.toolbar = self.addToolBar('退出')
        self.toolbar.addAction(exitAct)

        self.setGeometry(300, 300, 300, 200)
        self.setWindowTitle('工具栏')
        self.show()


if __name__ == "__main__":
    app = QApplication(sys.argv)
    ex = Example()
    sys.exit(app.exec_())

